src/services/Serialization.jl

```diff
@@ -84,7 +84,16 @@ function getTypeFromSerializationModule(softtypeString::String)
         else
             m = Main
         end
-        return getfield(m, Symbol(split_st[end]))        
+        noparams = split(split_st[end], r"{") 
+        ret = if 1 < length(noparams)
+            # fix #671, but does not work with specific module yet
```

Comment:
I think this constructor might work https://github.com/JuliaRobotics/IncrementalInference.jl/blob/4097cda7e0f6abcdf003e16c06f05e772b352874/src/Variables/DefaultVariables.jl#L46

Reply:
HI -- sure, we can improve that on update of serialization -- just need a solution for any VarType{T} at this point.